test/screenshot/lib/screenshot.js

```diff
@@ -48,18 +52,33 @@ const BROWSERS = [
 /** Map of URLs to `Progress` objects. */
 const progressMap = new Map();
 
+/**
+ * @type {!ParallelQueue<string>}
+ */
+const requestQueue = new ParallelQueue({maxParallels: API_PARALLEL_REQUEST_LIMIT});
+
 module.exports = {
   captureOneUrl,
 };
 
 async function captureOneUrl(testPageUrl) {
   logTestCaseProgress(testPageUrl, Progress.enqueued(BROWSERS.length));
 
-  return sendCaptureRequest(testPageUrl)
-    .then(
-      async (captureResponseBody) => handleCaptureResponse(testPageUrl, captureResponseBody),
-      async (err) => rejectWithError('captureOne', testPageUrl, err)
-    );
+  return requestQueue.enqueue(testPageUrl)
+    .then(() => {
+      return sendCaptureRequest(testPageUrl)
+        .then(
+          (captureResponseBody) => handleCaptureResponse(testPageUrl, captureResponseBody),
+          (err) => rejectWithError('captureOne', testPageUrl, err)
```

Comment:
Should probably dequeue the capture request in the error handlers unless the whole process exits on failure.

Reply:
Good catch! I was totally planning to do that, but clearly I forgot 😆 Thanks!!